src/js/components/TextInput/TextInput.js

```diff
@@ -121,6 +122,22 @@ const TextInput = forwardRef(
 
     const [focus, setFocus] = useState();
     const [showDrop, setShowDrop] = useState(false);
+    const [arrowDownInitiated, setArrowDownInitiated] = useState(false);
+
+    const usingKeyboard = useKeyboard();
+    useEffect(() => {
+      const handleKeyDown = (event) => {
+        if (event.key === 'ArrowDown') {
+          setArrowDownInitiated(true);
+        }
+      };
+
+      document.addEventListener('keydown', handleKeyDown);
```

Comment:
Shouldn't we be using Keyboard instead of directly doing window listeners like this, especially since this depends on events bubbling all the way to the window level